aten/src/ATen/native/quantized/cpu/qconv.cpp

```diff
@@ -335,6 +335,52 @@ class QConv2dInt8 final : public c10::OperatorKernel {
     // Force output format to be NHWC
     // TODO: consider preserving input format
     // TODO: add MemoryFormat::ChannelsLast here once perf is fixed
+    auto input_scale = input_contig.q_scale();
+
+    // Re-quantizing the bias based on input scale and weight scale.
+    if (!pack_ptr.input_scale.has_value()) {
+      // Get the original weight and adjust it to uint8 from int8
+      auto weight_contig = pack_ptr.orig_weight;
+      auto bias_fp32 = pack_ptr.bias;
+      int8_t* w_data = (int8_t*)weight_contig.data_ptr<c10::qint8>();
+      Tensor qnnp_weight = at::_empty_affine_quantized(
+          weight_contig.sizes(),
+          at::device(kCPU).dtype(kQUInt8),
+          kernel_scale,
+          kernel_zp);
+      auto* qnnp_w_data = qnnp_weight.data_ptr<c10::quint8>();
+      for (int i = 0; i < weight_contig.numel(); ++i) {
+        qnnp_w_data[i] = static_cast<c10::quint8>(w_data[i] + 128);
+      }
+      // Original bias was float, so we requantize it here.
+      auto bias = at::quantize_linear(
+          bias_fp32, kernel_scale * input_scale, 0, kQInt32);
+      // Update the input scale to not pack again.
+      pack_ptr.input_scale = input_scale;
+      auto wt_ptr =
+          guts::make_unique<PackedConvWeightsQnnp>(PackedConvWeightsQnnp{
```

Comment:
you don't need to create this struct again, do you? you can just assign the first member (PrePackConvWeights) and the `input_scale`

Reply:
You're right, I thought I had done it that way but apparently not :)